### ISSUE TITLE
How to train select agent?

### ISSUE THREAD

**Issue by cingtiye** [ShayanTalaei/CHESS#30]
Thank you for this great work! I'm every interested in your work. 

When I reproduce select agent model training, I faced some problems. I trained a binary classifier using BCE loss. My prompt is shown in the following picture. I tried to train LLaMA3.1 and Qwen2.5 in BIRD train set, but I failed. My model quickly converged on the training set, but overfited on the validation set.

Could you please tell me how you trained this model? and thanks in advance

![12162](https://github.com/user-attachments/assets/bafe3c26-6d01-4aa1-807b-c4cf1134b3a2)
